### PR TITLE
Don't run builds in CI on PR or push to main

### PR DESCRIPTION
**Pull request by ebsmothers**
Turns out that all our CI build jobs use PyTorch nightlies by default and we can't use stable PyTorch. It is pretty annoying to have CI fail on all our PRs all day whenever there is a breakage due to the nightlies. While not ideal, I am gonna disable our CI build jobs on PRs and push to main (until we can run them against stable PyTorch install). I claim this is relatively low-risk since P(no other job fails but build job fails due to one of our commits) << P(build job fails due to problem with PyTorch nightlies)

### CODE REVIEW

**Comment by joecummings** [comment]
Screenshot 2024-08-08 at 6 07 43 PM

**Comment by ebsmothers** [comment]
> Screenshot 2024-08-08 at 6 07 43 PM

just as i feared

**Comment by joecummings** [comment]
Try "release" @ebsmothers

File: .github/workflows/build_linux_wheels_stable.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    strategy:
+      fail-fast: false
+    with:
+      repository: pytorch/torchtune
+      ref: ""
+      package-name: torchtune
+      build-matrix: ${{ needs.generate-matrix.outputs.matrix }}
+      pre-script: .github/scripts/pre_build_script.sh
+      trigger-event: ${{ github.event_name }}
+      build-platform: 'python-build-package'
```
Comment (joecummings) [inline .github/workflows/build_linux_wheels_stable.yaml:43]: Add torchvision ya dingus

Comment (joecummings) [inline .github/workflows/build_linux_wheels_stable.yaml:43]:
```suggestion
      build-platform: 'python-build-package'
      pip-install-torch-extra-args: torchvision
```

**Comment by joecummings** [comment]
Not sure why this is now passing TBH - the workflow overrides the "release" command and installed nightly pytorch, which should fail with our version of torchao.

**Review by joecummings — APPROVED** [review]
go w the grace of god

**Pull request merged**